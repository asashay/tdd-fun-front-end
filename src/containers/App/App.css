.tdd-fun-app {
  background: #007991;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #78ffd6, #007991);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #78ffd6, #007991); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  height: 100vh;
  color: #ffffff;
}

.translated-text {
  background: lightgray;
}