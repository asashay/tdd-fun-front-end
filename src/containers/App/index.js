import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="d-flex flex-column tdd-fun-app">
        <div className="d-flex justify-content-end p-2">
          <button className="btn btn-warning">Login</button>
        </div>
        <div className="container flex-grow-1 d-flex flex-column justify-content-center align-items-center">
          <h1>Let's have some fun</h1>
          <p>Input text to translate it to master Yoda style</p>
          <div className="input-group">
            <textarea className="form-control" aria-label="With textarea" placeholder="Enter text..."></textarea>
          </div>
          <div className="my-2">
            <button className="btn btn-primary">Translate</button>
          </div>
          <div className="input-group">
            <textarea className="form-control" 
              aria-label="With textarea" 
              placeholder="Translated text..." 
              disabled></textarea>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
