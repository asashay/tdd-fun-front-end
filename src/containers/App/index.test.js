import React from 'react';
import {
  render,
  cleanup,
} from 'react-testing-library'

import App from './index';

afterEach(cleanup);

let app;
beforeEach(() => {
  app = render(<App />)
})

describe('App', () => {
  it('matches snapshot', () => {
    expect(app).toMatchSnapshot()
  });

  it('has login text', () => {
    expect(app.getByText('Login')).toBeTruthy()
  })

  it('has Let"s have some fun text', () => {
    expect(app.getByText("Let's have some fun")).toBeTruthy()
  })

  it('has "Input text to translate it to master Yoda style" text', () => {
    expect(app.getByText("Input text to translate it to master Yoda style")).toBeTruthy()
  })

  it('should have textarea to enter text', () => {
    expect(app.getByPlaceholderText("Enter text...")).toBeTruthy()
  })

  it('has translate button', () => {
    expect(app.getByText("Translate")).toBeTruthy()
  })

  it('has translated text area', () => {
    expect(app.getByPlaceholderText("Translated text...")).toBeTruthy()
  })
})
