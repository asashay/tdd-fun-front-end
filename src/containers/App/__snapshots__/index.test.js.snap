// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column tdd-fun-app"
      >
        <div
          class="d-flex justify-content-end p-2"
        >
          <button
            class="btn btn-warning"
          >
            Login
          </button>
        </div>
        <div
          class="container flex-grow-1 d-flex flex-column justify-content-center align-items-center"
        >
          <h1>
            Let's have some fun
          </h1>
          <p>
            Input text to translate it to master Yoda style
          </p>
          <div
            class="input-group"
          >
            <textarea
              aria-label="With textarea"
              class="form-control"
              placeholder="Enter text..."
            />
          </div>
          <div
            class="my-2"
          >
            <button
              class="btn btn-primary"
            >
              Translate
            </button>
          </div>
          <div
            class="input-group"
          >
            <textarea
              aria-label="With textarea"
              class="form-control"
              disabled=""
              placeholder="Translated text..."
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column tdd-fun-app"
    >
      <div
        class="d-flex justify-content-end p-2"
      >
        <button
          class="btn btn-warning"
        >
          Login
        </button>
      </div>
      <div
        class="container flex-grow-1 d-flex flex-column justify-content-center align-items-center"
      >
        <h1>
          Let's have some fun
        </h1>
        <p>
          Input text to translate it to master Yoda style
        </p>
        <div
          class="input-group"
        >
          <textarea
            aria-label="With textarea"
            class="form-control"
            placeholder="Enter text..."
          />
        </div>
        <div
          class="my-2"
        >
          <button
            class="btn btn-primary"
          >
            Translate
          </button>
        </div>
        <div
          class="input-group"
        >
          <textarea
            aria-label="With textarea"
            class="form-control"
            disabled=""
            placeholder="Translated text..."
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
